cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -106,9 +107,7 @@ private void deleteLocation(NamespaceId namespaceId) throws IOException {
         LOG.debug("Custom location mapping %s was found while deleting namespace %s. Deleting all data inside it but" +
                     "skipping namespace home directory delete.", namespaceHome, namespaceId);
         // delete everything inside the namespace home but not the namespace home as its user owned directory
-        for (Location childLocation : namespaceHome.list()) {
-          childLocation.delete(true);
-        }
+        Locations.deleteContent(namespaceHome);
```

Comment:
yes; DefaultNamespaceStorageHandler calls this method with impersonation, I believe.